Spencer-Damiano/Conceptual-Library
# Fix Login Page Centering and Add Profile Page with Dropdown Menu

# Fix Login Page Centering and Add Profile Page with Dropdown Menu

## Description

This pull request addresses the following:
1. Centers the login form elements on the login page.
2. Implements a basic profile page.
3. Updates the dropdown menu to include links to the profile page and handle user login/logout states.
4. Ensures global styles are applied throughout the application.

## Changes Made
- Updated `LoginPage.tsx` to include correct class names and imported global styles.
- Added `LoginPage.css` to style and center the login form elements.
- Created `ProfilePage.tsx` to display basic user information.
- Added `ProfilePage.css` for basic styling of the profile page.
- Updated `DropdownMenu.tsx` to include navigation links for the profile page and handle logout functionality.
- Imported global styles in `App.tsx` to ensure consistent styling across the application.

## To-Do List
- Go through the CSS files and implement a personal style guide for consistent and improved design. This will be done after ensuring core functionality is complete.

## How to Test
1. Pull this branch to your local environment.
2. Run `npm install` to ensure all dependencies are installed.
3. Start the development server with `npm start`.
4. Navigate to `/login` and check that the login form is centered.
5. Log in with a test user and verify that you are redirected to the dashboard.
6. Use the dropdown menu to navigate to the profile page and check that it displays user information.
7. Test the logout functionality to ensure it redirects to the login page.

## Checklist
- [x] Center login form elements.
- [x] Create and style the profile page.
- [x] Update dropdown menu with profile link and logout functionality.
- [x] Apply global styles across the application.
- [ ] Implement personal CSS style guide (future task).

## Related Issues
- Closes #1: Add a to-do list to implement personal CSS style guide.

## Screenshots (if applicable)
Add screenshots here if they are helpful for the review.

## Notes
This is a personal project, so the pull request is self-reviewed. Ensure thorough testing before merging into the main branch.
